common/url_util.cc

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/common/extensions/extension_constants.h"
+#include "brave/common/url_util.h"
+#include "url/gurl.h"
+
+namespace brave {
+
+GURL GetURLOrPDFURL(const GURL& url) {
+  if (url.SchemeIs("chrome-extension") &&
+      url.host() == pdfjs_extension_id) {
```

Comment:
[minor] it would be helpful to set `chrome-extension://<pdfjs_extension_id>/` as a constant ` pdfjs_extension_prefix` in extension_constants since it is repeated a lot in this PR and will also be needed in the PR to hide the pdfjs prefix.

then this check can be replaced with `if (url.GetOrigin() == pdfjs_extension_prefix)`